canvas_exercise/shapes_game/index.js

```diff
@@ -1,36 +1,130 @@
 window.addEventListener("load", function() {
 
-  function clear(ctx, width, heigt) {
+  function clear(ctx, width, height) {
+    ctx.clearRect(0, 0, width, height);
   }
 
   function drawRandomShape(ctx, width, height) {
+    var randomShapeNumber = Math.floor(Math.random()*(4)+1);
+    var x = Math.floor(Math.random()*642); // multiplied by 692 to ensure shape doesn't go off black screen 
+    var y = Math.floor(Math.random()*592); // multiplied by 642 to ensure shape doesn't go off black screen
+
+    // 1 - White Triangle
+    if (randomShapeNumber === 1) {
+      expectedKey = "white0";
+      ctx.fillStyle = "white";
+      ctx.beginPath();
```

Comment:
Lots of duplicate code between these two if statements.  I'd try to consolidate it and only change the color based on red or white triangle